kmip/pie/client.py

```diff
@@ -627,18 +631,21 @@ def mac(self, uid, algorithm, data):
             message = result.result_message.value
             raise exceptions.KmipOperationFailure(status, reason, message)
 
-    def _build_key_attributes(self, algorithm, length):
+    def _build_key_attributes(self, algorithm, length, masks=list()):
         # Build a list of core key attributes.
         algorithm_attribute = self.attribute_factory.create_attribute(
             enums.AttributeType.CRYPTOGRAPHIC_ALGORITHM,
             algorithm)
         length_attribute = self.attribute_factory.create_attribute(
             enums.AttributeType.CRYPTOGRAPHIC_LENGTH,
             length)
+        # Default crypto usage mask value
+        mask_value = [enums.CryptographicUsageMask.ENCRYPT,
+                      enums.CryptographicUsageMask.DECRYPT]
+        mask_value.extend(masks)
```

Comment:
We should probably remove any duplicates in the list here. It shouldn't really matter since the flags get combined latter on, but I'd prefer to keep things tidy as we go.